dvc/repo/diff.py

```diff
@@ -101,6 +101,9 @@ def _to_checksum(output):
     for stage in repo.stages:
         for output in stage.outs:
             if _exists(output):
+                # if a target was supplied, filter out non-target files
+                if target is not None and not str(output).startswith(target):
```

Comment:
startswith is susceptible to a bug when you don't take into account the path separator. Also `str(output)` is not the best way to get an output path. Please check out `output.path_info`.

Reply:
Addressed in @efiop makes sense, incorporated in fa6c3b4d3868b240c12200c8dc8082e1bfc4ab3b.